backend/amd64/proc.ml

```diff
@@ -409,6 +410,7 @@ let op_is_pure = function
   | Icall_ind | Icall_imm _ | Itailcall_ind | Itailcall_imm _
   | Iextcall _ | Istackoffset _ | Istore _ | Ialloc _
   | Iintop(Icheckbound) | Iintop_imm(Icheckbound, _) | Iopaque -> false
+  | Ispecific(Ipause)
```

Comment:
It is not immediately clear to me why it is not pure.

Reply:
Or is it only to ensure the instruction will never be
moved or optimized away?

Reply:
exactly, and to prevent it from being optimized away.